test/e2e/case1_framework_deployment_test.go

```diff
@@ -264,9 +269,61 @@ var _ = Describe("Test framework deployment", func() {
 			GetWithTimeoutClusterResource(cluster.clusterClient, gvrNamespace, cluster.clusterName, false, 15)
 		}
 	})
+
+	It("should deploy with startupProbes or initialDelaySeconds depending on version", func() {
+		// Note that these regex only work well for k8s versions 1.10 - 1.29
+		modernK8sRegex := regexp.MustCompile(`Server Version: v1\.2`)
+		legacyK8sRegex := regexp.MustCompile(`Server Version: v1\.1`)
+
+		for i, cluster := range managedClusterList[1:] {
+			Expect(cluster.clusterType).To(Equal("managed"))
+
+			logPrefix := cluster.clusterType + " " + cluster.clusterName + ": "
+			By(logPrefix + "deploying the default framework managedclusteraddon")
+			Kubectl("apply", "-n", cluster.clusterName, "-f", case1ManagedClusterAddOnCR)
+			deploy := GetWithTimeout(
+				cluster.clusterClient, gvrDeployment, case1DeploymentName, addonNamespace, true, 30,
+			)
+			Expect(deploy).NotTo(BeNil())
+
+			containers, found, err := unstructured.NestedSlice(deploy.Object, "spec", "template", "spec", "containers")
+			Expect(err).To(BeNil())
+			Expect(found).To(BeTrue())
+
+			container, ok := containers[0].(map[string]interface{})
+			Expect(ok).To(BeTrue())
+
+			version := Kubectl(
+				"version",
+				"--short",
+				fmt.Sprintf("--kubeconfig=%s%d.kubeconfig", kubeconfigFilename, i+1),
+			)
+
+			if modernK8sRegex.MatchString(version) {
```

Comment:
Right now, if neither matches, we can fail the test and indicate that we need to update the regex. If it's important to you, I can make a more robust regex now, but I wanted to keep it simple initially

Reply:
Maybe it'd just be better to parse the semver with https://github.com/Masterminds/semver so you can do a greater than statement rather than rely on regex.